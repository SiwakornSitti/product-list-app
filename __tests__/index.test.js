// Import มาแต่ไม่ได้ใช้ 
import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import ProductList from '../components/product-list.js'
import ProductDetail from "../components/product-detail";

const mockProducts = [
    { 
        "id": 2,    
        "name": "b00021xhzw", 
        "number": "adobe after effects professional 6.5 upgrade from standard to professional", 
        "description": "upgrade only; installation of after effects standard new disk caching tools speed up your interactive work save any combination of animation parameters as presets", 
        "images": []  
    },
    { 
        "id": 3,    
        "name": "b00021xhzw", 
        "number": "domino designer/developer v5.0", 
        "description": "reference domino designer/developer r5 doc pack includes the following titles: application development with domino designer (intermediate-advanced) 536 pages it explains building applications creating databases using forms fields views folders navi", 
        "images": [
            { 
            "url": "http://lorempixel.com/200/200/people/", 
            "name": "cover"
            }]
    }
]

describe('With Snapshot Testing', () => {
    // ควรมีเทสการ Render แต่ไม่พบข้อมูลด้วย 
    // อยากให้เขียนเทสที่ไม่ใช้แค่ Snapshot มันเหมือนไว้แค่ Detect change อย่างเดียว  
 
    it('Render ProductItem', () => {
        const component = renderer.create(<ProductList products={mockProducts} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    
    it('Render ProductDetail', () => {
        let product  = mockProducts[1]
        const component = renderer.create(<ProductDetail product={product}/>)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
