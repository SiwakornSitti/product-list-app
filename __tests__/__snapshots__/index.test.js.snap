// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Render Product Item 1`] = `
Array [
  <li
    className="sc-ifAKCX haqxAK"
  >
    <a
      className="sc-bZQynM bEwXTb"
      onClick={[Function]}
    >
      b00021xhzw
       - 
      adobe after effects professional 6.5 upgrade from standard to professional
    </a>
  </li>,
  <li
    className="sc-ifAKCX haqxAK"
  >
    <a
      className="sc-bZQynM bEwXTb"
      onClick={[Function]}
    >
      b00021xhzw
       - 
      domino designer/developer v5.0
    </a>
  </li>,
]
`;

exports[`With Snapshot Testing Render ProductDetail 1`] = `
<div>
  <h1>
    b00021xhzw
  </h1>
  <span>
    domino designer/developer v5.0
  </span>
  <p>
    reference domino designer/developer r5 doc pack includes the following titles: application development with domino designer (intermediate-advanced) 536 pages it explains building applications creating databases using forms fields views folders navi
  </p>
  <div>
    <div>
      <img
        alt="cover"
        className="sc-EHOje gZcETi"
        src="http://lorempixel.com/200/200/people/"
      />
      <span>
        cover
      </span>
    </div>
  </div>
</div>
`;

exports[`With Snapshot Testing Render ProductItem 1`] = `
Array [
  <li
    className="sc-ifAKCX haqxAK"
  >
    <a
      className="sc-bZQynM bEwXTb"
      onClick={[Function]}
    >
      b00021xhzw
       - 
      adobe after effects professional 6.5 upgrade from standard to professional
    </a>
  </li>,
  <li
    className="sc-ifAKCX haqxAK"
  >
    <a
      className="sc-bZQynM bEwXTb"
      onClick={[Function]}
    >
      b00021xhzw
       - 
      domino designer/developer v5.0
    </a>
  </li>,
]
`;
